src/binary_basic_decode.ml

```diff
@@ -0,0 +1,640 @@
+open Types
+open Syntax
+
+module Input = struct
+  type t =
+    { bytes : string
+    ; pt : int
+    ; size : int
+    }
+
+  let _pp input =
+    print_int input.pt;
+    print_string " ";
+    print_int input.size;
+    print_newline ()
+
+  let is_empty input = input.size = 0
+
+  let from_bytes bytes =
```

Comment:
awkward "bytes" when it is actually a string

Reply:
Keeping `bytes` in some function names seems to make sense: it refers to the binary format of the source file.